src/TraceEvent/TraceEventStacks.cs

```diff
@@ -758,14 +761,18 @@ public StackSourceCallStackIndex GetCallStack(CallStackIndex callStackIndex, Sta
             {
                 callerIdx = top;
 
-                var frameName = GetFrameName(frameIdx, false);
-                var bangIdx = frameName.IndexOf('!');
-                if (0 < bangIdx)
+                // TODO Fix NOW Clr thread sample only has managed call stack. Do we have a better solution to differentiate it from normal call stack?
+                if (!string.Equals(m_log.origin, EventPipeEventSource.EventPipe, StringComparison.OrdinalIgnoreCase))
```

Comment:
We should make up a property onlyManagedCodeStacks for this purpose instead of origin.